### DISCUSSION TITLE
Confusion on Listing 7.27

### DISCUSSION THREAD

**Discussion by joelf1403** [IntelliTect/EssentialCSharp.Web#370] [Q&A] [answered]
The text says this: "In addition, Listing 7.27 assigns the third element in the tuple to the variable fileName if the initial match expression evaluates to true.". But from what I see the tuple has only 2 elements, not 3. And fileName is a constant int. Can someone help me understand this? I am a longtime C++ pro but new to C#, so go easy on me if I am just not getting it!

**Comment by BenjaminMichaelis** [comment, marked as answer]
@joelf1403 Thank you for pointing this out!

You’re absolutely right—the tuple contains only two elements, and the variable fileName serves as a constant integer, used as an index to access the second element from the args array in the encrypt command case. It isn’t assigned any value from the tuple.

To address this issue, we’ve updated the sentence to be as follows:

> In addition, Listing 7.27 retrieves the filename from the second item in args if the match expression evaluates to true.

This revised content will be reflected on our website in the coming days. If you have any further feedback or questions, feel free to reach out. We appreciate your engagement with our material! 😊
